.github/workflows/inspektor-gadget.yml

```diff
@@ -2,7 +2,7 @@ name: Inspektor Gadget CI
 env:
   REGISTRY: ghcr.io
   CONTAINER_REPO: ${{ github.repository }}
-  GO_VERSION: 1.20.10
```

Comment:
Why is this upgrade necessary for this PR?

Reply:
To have `slices.Clone()`, I mean looking at what `slices.Clone()` we can also do (using [append](https://stackoverflow.com/questions/71316774/is-it-equivalent-use-copy-or-append-slice-type-in-golang)):

```args := append([]string{}, os.Args[1:]...) // clone to avoid modifying the original os.Args```

But should be fine to go to `1.21`. What do you say?

Reply:
IMO, I would avoid upgrading it unless it is strictly necessary or it provides a big advantage. And, I consider this doesn't seem the case.

However, let's see what other maintainers think.

Reply:
I think 1.21 with its `maps` and `slices` packages and the new `min()` and `max()` funcs alone are worth the upgrade as we use those things constantly. And this is timed just after a release, so there's plenty of time ahead of us to make sure it's running well. So my opinion is: go for it.

The release notes state that Go 1.21 no longer runs on Win 7 & 8 and older MacOS versions, but I don't know if the binaries it produces are affected as well - we should check for that and maybe add that to the release notes.

Reply:
> (But maybe this should get its own PR prior to this one)

Yeah, maybe it is a better idea to open a separate PR for golang 1.21 and gather more use-cases.

> `args := append([]string{}, os.Args[1:]...) // clone to avoid modifying the original os.Args`

If the [above](https://github.com/inspektor-gadget/inspektor-gadget/pull/2195/files#r1381893852) approach looks good, I can unblock this fix and we can continue with `golang 1.21` separately?

Reply:
I'll suggest not to tie these two things together. Unblock this PR by using that approach and add a comment saying it should be changed once we use go1.21, then in another PR we can discuss the go1.21 implications.

Reply:
Makes sense. I would open a separate PR for bumping `golang` version. For now I switched to different approach with a comment.

Reply:
I opened https://github.com/inspektor-gadget/inspektor-gadget/pull/2206 to update golang version.